lib/browser/guest-view-manager.js

```diff
@@ -149,7 +149,8 @@ const createGuest = function (embedder, params) {
   // Forward internal web contents event to embedder to handle
   // native window.open setup
   guest.on('-add-new-contents', (...args) => {
-    if (guest.getLastWebPreferences().nativeWindowOpen === true) {
+    const webPreferences = guest.getLastWebPreferences()
+    if (webPreferences.nativeWindowOpen === true || webPreferences.sandbox === true) {
```

Comment:
Does `sandbox` truly imply `nativeWindowOpen`?  I think we have checks for `nativeWindowOpen` in a few places around the code base.  We should probably adjust the web prefs impl to set `nativeWindowOpen` to `true` if `sandbox` is `true`

Reply:
We don't execute `window-setup` in sandbox. The other places are: https://github.com/electron/electron/commit/aa241771a381560e586456850119245cbdca2c9e

Reply:
we have comments like
> "window.open", used in sandbox and nativeWindowOpen mode.